lambdas/services/edge_presign_service.py

```diff
@@ -40,13 +55,43 @@ def attempt_url_update(self, uri_hash, origin_url) -> None:
             logger.error(f"{str(e)}", {"Result": LambdaError.EdgeNoClient.to_str()})
             raise CloudFrontEdgeException(400, LambdaError.EdgeNoClient)
 
-    def extract_environment_from_url(self, url: str) -> str:
-        match = re.search(r"https://([^.]+)\.[^.]+\.[^.]+\.[^.]+", url)
+    @staticmethod
+    def update_s3_headers(request: dict, request_values: dict):
+        domain_name = request_values["domain_name"]
+        if "authorization" in request["headers"]:
+            del request["headers"]["authorization"]
+        request["headers"]["host"] = [{"key": "Host", "value": domain_name}]
+
+        return request
+
+    @staticmethod
+    def filter_request_values(request: dict) -> dict:
+        try:
+            uri: str = request["uri"]
+            querystring: str = request["querystring"]
+            headers: dict = request["headers"]
+            origin: str = request.get("origin", {})
+            domain_name: str = origin["s3"]["domainName"]
+        except KeyError as e:
+            logger.error(f"Missing request component: {str(e)}")
+            raise CloudFrontEdgeException(500, LambdaError.EdgeNoOrigin)
+
+        return {
+            "uri": uri,
+            "querystring": querystring,
+            "headers": headers,
+            "domain_name": domain_name,
+        }
+
+    @staticmethod
+    def filter_domain_for_env(domain_name: str) -> str:
```

Comment:
Does this filter take `prod` into account? as there is no environment to extract from the domain